src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +441,331 @@ private static String trimExcessWhitespaces(String text) {
+        }
+        return result.toString().trim();
+    }
+
+    /**
+     * Get Description node from ast.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode descriptionNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                for (DetailNode dataType : node.getChildren()) {
+                    if (dataType.getType() == JavadocTokenTypes.DESCRIPTION) {
+                        descriptionNode = dataType;
+                    }
+                }
+            }
+        }
+        return descriptionNode;
+    }
+
+    /**
+     * Finds and returns first sentence.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineFirstSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                result.append(returnInlineFirstSentence(node));
+            }
+
+            if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                result.append(returnInlineFirstSentence(
+                        Objects.requireNonNull(getInlineTagNodeInsideHtml(ast))));
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Sets result string in method getInlineFirstSentence.
+     *
+     * @param node JavaDoc Inline node.
+     * @return first sentence.
+     */
+    public static String returnInlineFirstSentence(DetailNode node) {
+        final StringBuilder result = new StringBuilder(256);
+        final String periodSuffix = PERIOD;
+        for (DetailNode child : node.getChildren()) {
+            if (ALLOWED_TYPES.contains(child.getType())) {
+                final String text;
+                if (child.getChildren().length == 0) {
+                    text = child.getText();
+                }
+                else {
+                    text = getTextFromDescription(child);
+                }
+
+                if (text.contains(periodSuffix)) {
+                    result.append(text, 0, text.indexOf(periodSuffix) + 1);
+                    break;
+                }
+
+                result.append(text);
+            }
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Returns the text from an InlineTag Description.
+     *
+     * @param child InlineTag Node.
+     * @return text of InlineTag Description.
+     */
+    private static String getTextFromDescription(DetailNode child) {
+        final String period = PERIOD + ' ';
+        final DetailNode[] child2 = child.getChildren();
+        StringBuilder inlineText = null;
+        for (DetailNode ch2 : child2) {
+            if (ch2.getType() == JavadocTokenTypes.TEXT) {
```

Comment:
please fix all variables with numbers, there should be no numbers in names

Reply:
Done